updated the Custom Callback Configuration provider (#3678)
meshery/meshery: docs/pages/extensibility/providers.md

}

```
Meshery enables you as a service mesh owner to customize your service mesh deployment.

> ## Meshery Server supports for custom callback URL for connected Remote Providers

Reviewer: ```suggestion
## Configurable OAuth Callback URL
```

> As of release v0.5.39, Meshery Server has extended its capability to support callback URL for connected Remote Providers. It enables user to define there own redirect endpoints for the connected remote provider. This is helpful for the user who are deploying meshery behind the reverse proxy or ingress gateway.

Reviewer: ```suggestion
As of release v0.5.39, Meshery Server has extended its capability to configure the callback URL for connected Remote Providers. Helpful when you are deploying Meshery behind an ingress gateway or reverse proxy. You can specify a custom, redirect endpoints for the connected Remote Provider. 
```

**Sample UseCase:**

> User has deployed the Meshery behind a Istio ingressgateway and the Istio is also behind a Application Gateway (in a cloud infrastructure). Generally when you use github remote provider for authentication, it redirect the request to ingressgateway FQDN. But in this setup the redirection won't be successful becuase the ingress gateway is behind an addition application Gateway. In this case you have to define where the request should be redirected once it is authenticated from github.

Reviewer: ```suggestion
User has deployed the Meshery behind a Istio Ingress Gateway and the Istio is also behind an Application Gateway (e.g. AKS Application Gateway). Generally, when you use a GitHub Remote Provider for authentication, it redirect the request to the Istio Ingress Gateway FQDN. In this setup, redirection won't be successful because the Ingress Gateway is behind an additional Application Gateway. In this case, you have to define where the request should be redirected once it is authenticated from GitHub.
```

> ***Solution:***

Reviewer: #### Configuring MESHERY_SERVER_CALLBACK_URL

> You can define your custom callback URL by setting up `MESHERY_SERVER_CALLBACK_URL` env variable while installing meshery in your K8s cluster.

Reviewer: ```suggestion
You can define your custom callback URL by setting up the `MESHERY_SERVER_CALLBACK_URL` environment variable while deploying Meshery in your Kubernetes cluster.
```

> Example:

Reviewer: ```suggestion
```

> If you are deploying meshery through HELM and your Meshery is expected to be accessed through URL `https://k8s-staging.test.io/`, then the command is:

Reviewer: ```suggestion
If you are deploying Meshery using Helm, you can configure the MESHERY_SERVER_CALLBACK_URL as shown in the following example.

- **Custom URL:** `https://k8s-staging.test.io/`
- **Auth Endpoint:** `api/user/token` (append at the end of your custom URL)
```

```
helm install meshery meshery/meshery --namespace meshery --set env.MESHERY_SERVER_CALLBACK_URL=https://k8s-staging.test.io/api/user/token
```

> points to remember here:

Reviewer: ```suggestion
##### Note
```

> - for user authentication, Meshery expect `api/user/token` to be appended at the end of your custom URL

Reviewer: ```suggestion
```

> - if you are setting up meshery with proxy and accessing it via path (like: `https://k8s-staging.test.io/meshery` ) then your callback URL would be `https://k8s-staging.test.io/meshery/user/token`

Reviewer 2: Maybe we need this?
```diff
-  `https://k8s-staging.test.io/meshery/user/token`
+ `https://k8s-staging.test.io/meshery/api/user/token`
```

Author: Ahh .. Yes , Sorry .. I will add that

> - if you are setting up meshery with proxy and accessing it via path (like: `https://k8s-staging.test.io/meshery` ) then your callback URL would be `https://k8s-staging.test.io/meshery/api/user/token`

Reviewer: ```suggestion
With a path like, `https://k8s-staging.test.io/meshery`, your callback URL will be `https://k8s-staging.test.io/meshery/api/user/token`. 
```
